polar-core/src/partial/partial.rs

```diff
@@ -130,23 +130,25 @@ impl Partial {
         Term::new_temporary(Value::Partial(Partial::new(name)))
     }
 
-    pub fn compare(&mut self, operator: Operator, other: Term) {
-        assert!(matches!(
-            operator,
-            Operator::Lt
-                | Operator::Gt
-                | Operator::Leq
-                | Operator::Geq
-                | Operator::Eq
-                | Operator::Neq
-        ));
-
-        let op = Operation {
-            operator,
-            args: vec![self.variable_term(), other],
-        };
-
-        self.add_constraint(op);
+    pub fn compare(&mut self, operator: Operator, operand: Operand) {
+        use Operator::{Eq, Geq, Gt, Leq, Lt, Neq};
+        let asymmetric_op = matches!(operator, Gt | Geq | Lt | Leq);
+        let symmetric_op = matches!(operator, Eq | Neq);
+        assert!(asymmetric_op || symmetric_op);
+
+        // Normalize comparison operations so this is always on LHS.
+        let args = vec![self.variable_term(), operand.term];
+        let mut operation = Operation { operator, args };
+        if operand.side == Side::Left && asymmetric_op {
+            operation.operator = match operation.operator {
+                Gt => Lt,
+                Geq => Leq,
+                Lt => Gt,
+                Leq => Geq,
+                o => o,
```

Comment:
Should this final case be `unreachable!()`? Due to L135.

Reply:
091ad77